### PR TITLE
Sign up validations

### PR DESCRIPTION
**Pull request by jonnyalexbh**
## Summary
- sign up validations

## Trello Card
https://trello.com/c/uJSxraY7/24-sign-up-validaciones

## Screenshot
Captura de pantalla 2019-08-12 a la(s) 2 03 02 p  m

### CODE REVIEW

File: app/graphql/inputs.js [hunk 1]
```diff
@@ -5,8 +5,8 @@ module.exports = gql`
     firstName: String!
     lastName: String!
     username: String!
-    email: String!
-    password: String!
+    email: String! @constraint(format: "email", minLength: 12, pattern: ".*@wolox.(ar|co|cl|com.ar)$")
+    password: String! @constraint(minLength: 8)
```
Comment (emanuelcasco) [inline app/graphql/inputs.js]: password should be alphanumeric, it should throw error for special characters as: `_*'?`

File: app/helpers.js [hunk 2]
```diff
@@ -1,4 +1,10 @@
-const { MAP_ORDER } = require('./constants');
+const bcrypt = require('bcryptjs'),
+  salt_sync = require('../config').encryption,
+  { MAP_ORDER } = require('./constants');
+
+const salt = bcrypt.genSaltSync(parseInt(salt_sync));
+
+exports.encryptPassword = password => bcrypt.hashSync(password, salt);
```
Comment (emanuelcasco) [inline app/helpers.js]: Here you are executing this task synchronously. You can implement in asynchronously :)

**Review by emanuelcasco — CHANGES_REQUESTED** [review]

File: app/graphql/inputs.js (hunk 1 above)
Comment (jonnyalexbh) [reply app/graphql/inputs.js]: I forgot it

File: app/graphql/users/resolver.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const { user: User } = require('../../models'),
+  errors = require('../../errors'),
+  logger = require('../../logger'),
+  helpers = require('../../helpers');
+
+exports.createUser = async user => {
+  try {
+    const hash = await helpers.encryptPassword(user.password);
+    user.password = hash;
```
Comment (emanuelcasco) [inline app/graphql/users/resolver.js]: assign it directly: `user.password = await helpers.encryptPassword(user.password);`

File: app/graphql/users/resolver.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    const storeUser = await User.create(user);
```
Comment (emanuelcasco) [inline app/graphql/users/resolver.js]: storedUser or savedUser, variable name shouldn't be a verb or action

File: app/helpers.js [hunk 5]
```diff
@@ -1,4 +1,11 @@
-const { MAP_ORDER } = require('./constants');
+const bcrypt = require('bcryptjs'),
+  salt_sync = require('../config').encryption,
+  { MAP_ORDER } = require('./constants');
+
+exports.encryptPassword = async password => {
+  const salt = await bcrypt.genSalt(salt_sync);
```
Comment (emanuelcasco) [inline app/helpers.js]: Is this working? According to documentation [async functions require a callback](https://www.npmjs.com/package/bcryptjs#gensaltrounds-seed_length-callback)

Comment (emanuelcasco) [reply app/helpers.js]: In order to promisify a callback function you can wrap it into a Promise or use [Util.promisify](https://nodejs.org/dist/latest-v8.x/docs/api/util.html) package from node

**Review by emanuelcasco — CHANGES_REQUESTED** [review]

File: app/graphql/albums/resolvers.js [hunk 6]
```diff
@@ -5,20 +5,26 @@ exports.getAlbum = async (_, params) => {
   const data = {
     ...album,
     artist: album.userId,
-    name: album.title,
-    photos: await photosByAlbumId(params.id)
+    name: album.title
```
Comment (emanuelcasco) [inline app/graphql/albums/resolvers.js:8]: Album shouldn't have field `name`

**Review by emanuelcasco — CHANGES_REQUESTED** [review]

File: app/graphql/albums/resolvers.js (hunk 6 above)
Comment (jonnyalexbh) [reply app/graphql/albums/resolvers.js:8]: will be corrected in https://github.com/wolox-training/jb-graphql/pull/7#discussion_r313487397

**Review by emanuelcasco — APPROVED** [review]

**Pull request merged**